File: sdk-dotnet/UsageBreakdown.cs
using System.Text.Json.Serialization;

namespace Tollara;

/// <summary>Shared usage breakdown for estimate and report responses (see docs-sdk/MAIN-SDK-API-SPEC.md).</summary>
public record UsageBreakdown
{
    public decimal? UnitsUsed { get; init; }
    public decimal? BaseUnitsUsed { get; init; }
    public decimal? OverageUnits { get; init; }
    public decimal? ChargeableOverageUnits { get; init; }
    public decimal? SurplusOverageUnits { get; init; }
    public decimal? OverageCost { get; init; }
    public decimal? TotalOverageCost { get; init; }
    public decimal? UnitsRemaining { get; init; }

    /// <summary>PREPAID: credit balance after this chunk; null for other billing models.</summary>
    public decimal? RemainingCredits { get; init; }

    public decimal? RemainingSpendingCap { get; init; }
    public decimal? TotalUnitsUsedThisCycle { get; init; }

    [JsonPropertyName("isOverLimit")]
    public bool? OverLimit { get; init; }

    [JsonPropertyName("isOverage")]
    public bool? Overage { get; init; }

    [JsonPropertyName("isOverageAllowed")]
    public bool? OverageAllowed { get; init; }
}

File: sdk-java/src/main/java/com/tollara/client/model/UsageBreakdown.java
package com.tollara.client.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

/**
 * Shared usage breakdown for estimate and report responses (see docs-sdk/MAIN-SDK-API-SPEC.md).
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
public class UsageBreakdown {

    private BigDecimal unitsUsed;
    private BigDecimal baseUnitsUsed;
    private BigDecimal overageUnits;
    private BigDecimal chargeableOverageUnits;
    private BigDecimal surplusOverageUnits;
    private BigDecimal overageCost;
    private BigDecimal totalOverageCost;
    private BigDecimal unitsRemaining;
    /** PREPAID: credit balance after this chunk; null for other billing models. */
    private BigDecimal remainingCredits;
    private BigDecimal remainingSpendingCap;
    private BigDecimal totalUnitsUsedThisCycle;

    @JsonProperty("isOverLimit")
    private Boolean overLimit;

    @JsonProperty("isOverage")
    private Boolean overage;

    @JsonProperty("isOverageAllowed")
    private Boolean overageAllowed;
}
